llama_index/embeddings/openai.py

```diff
@@ -103,6 +104,13 @@ class OpenAIEmbeddingModeModel(str, Enum):
 @retry(
     wait=wait_random_exponential(min=1, max=20),
     stop=stop_all(stop_after_attempt(6), stop_after_delay(60)),
+    retry=(
```

Comment:
thanks @bhavishpahwa for contribution.
Just wondering could you reuse `_create_retry_decorator` in https://github.com/run-llama/llama_index/blob/main/llama_index/llms/openai_utils.py#L110-L127

Reply:
Well I initially wanted to reuse that but I thought there might be a reason why initially here `wait_random_exponential` was used instead of `wait_exponential` used in the original `_create_retry_decorator`. Although I think it is better to use wait_exponential as it is better to use in a usecase when one is waiting for an unavailable resource to become available again. Also originally `stop_after_delay(60)` is used which is cool as it means that the retry operation would be only attempted for a maximum of 60 secs and this is important as most of the vector index formation relies on these methods.

Reply:
ic thanks for explanation. 
then could we somehow refactor `_create_retry_decorator` to adapt `wait_random_exponential` as an option. so that we could resue the same _create_retry_decorator function here. There would be many duplicated retry logic in this file.

Reply:
thanks @bhavishpahwa !!!

Reply:
@hatianzhang had pushed the updated code can you review.